atom/renderer/api/atom_api_web_frame.cc

```diff
@@ -368,6 +374,102 @@ void WebFrame::ClearCache(v8::Isolate* isolate) {
     base::MemoryPressureListener::MEMORY_PRESSURE_LEVEL_CRITICAL);
 }
 
+v8::Local<v8::Value> WebFrame::Opener() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->Opener()->ToWebLocalFrame();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::Parent() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->Parent()->ToWebLocalFrame();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::Top() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->Top()->ToWebLocalFrame();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::FirstChild() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->FirstChild()->ToWebLocalFrame();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::NextSibling() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->NextSibling()->ToWebLocalFrame();
+  if (local_frame)
+    return mate::CreateHandle(isolate(),
+                              new WebFrame(isolate(), local_frame)).ToV8();
+  else
+    return v8::Null(isolate());
+}
+
+v8::Local<v8::Value> WebFrame::TraverseNext() const {
+  blink::WebLocalFrame* local_frame =
+    WebFrame::web_frame_->TraverseNext()->ToWebLocalFrame();
```

Comment:
The travel would stop when encountered a remote frame. Also `TraverseNext` returns nullptr at the end so it would then crash here.

If we do not really need this API, I prefer to simply remove this.

Reply:
I included it just because it was there, so I am ok with removing it, firstChild and nextSibling should enough to walk the whole tree.